pkg/apis/pingcap.com/v1alpha1/types.go

```diff
@@ -103,26 +103,28 @@ type TidbClusterStatus struct {
 	TiDB      TiDBStatus `json:"tidb,omitempty"`
 }
 
+// PodSpec repreresents shared pod fields between PD/TiDB/TiKV
+type PodSpec struct {
+	Replicas     int32               `json:"replicas"`
+	Affinity     *corev1.Affinity    `json:"affinity,omitempty"`
+	NodeSelector map[string]string   `json:"nodeSelector,omitempty"`
+	Tolerations  []corev1.Toleration `json:"tolerations,omitempty"`
+	Annotations  map[string]string   `json:"annotations,omitempty"`
+	HostNetwork  bool                `json:"hostNetwork,omitempty"`
+}
```

Comment:
`PodSpec` is created to simplify code only. Only one field `hostNetwork` is introduced.

Reply:
This has been reverted because it cannot pass `go vet` check.

```
pkg/apis/pingcap.com/v1alpha1/tidbcluster_test.go:180:5: cannot use promoted field PodSpec.Replicas in struct literal of type PDSpec
pkg/apis/pingcap.com/v1alpha1/tidbcluster_test.go:183:5: cannot use promoted field PodSpec.Replicas in struct literal of type TiKVSpec
pkg/apis/pingcap.com/v1alpha1/tidbcluster_test.go:186:5: cannot use promoted field PodSpec.Replicas in struct literal of type TiDBSpec
...
```